### PR TITLE
Update docs for turbot_resource

### PR DESCRIPTION
**Pull request by RupeshPatil20**

### CODE REVIEW

File: website/docs/r/resource.html.markdown [hunk 1]
```diff
@@ -35,14 +35,46 @@ EOT
...
+```
+
 ## Argument Reference
 
 The following arguments are supported:
 
 - `parent` - (Required) The `id` or `aka` of the level at which the Turbot resource will be created.
 - `type` - (Required) Defines the type of the resource to be created.
-- `data` - (Required) JSON representation of the details of the resource. When parsed, it must be valid for the `type` schema.
+- `data` - (Required) JSON representation of the details of the resource, managed on merge basis. When parsed, it must be valid for the `type` schema.
```
Comment (kaidaguerre) [inline website/docs/r/resource.html.markdown]: `data` - (Optional) JSON representation of resource properties to be managed by Terraform. These must be valid for the resource`type` schema. NOTE: If additional properties are set on the resource by other means, they are ignored by Terraform. 
`metadata` - (Optional) JSON representation of resource metadata properties to be managed by Terraform. NOTE: If additional metadata properties are set on the resource by other means, they are ignored by Terraform. 
`full_data` - (Optional) JSON representation of all resource properties to be set on the resource. These must be valid for the resource`type` schema. NOTE: If additional properties are set on the resource by other means, they are removed. 
`full_metadata` - (Optional) JSON representation of all resource metadata properties to be set on the resource. NOTE: If additional metadata properties are set on the resource by other means, they are removed.

File: website/docs/r/resource.html.markdown [hunk 2]
```diff
@@ -35,14 +35,46 @@ EOT
...
+}
+EOT
+}
+```
+
 ## Argument Reference
 
 The following arguments are supported:
 
 - `parent` - (Required) The `id` or `aka` of the level at which the Turbot resource will be created.
```
Comment (kaidaguerre) [inline website/docs/r/resource.html.markdown]: - `parent` - (Required) The identifier of the parent resource under which this resource will be created.

File: website/docs/r/resource.html.markdown [hunk 3]
```diff
@@ -35,14 +35,46 @@ EOT
...
 
+###Using full_data
+
+```hcl
+resource "turbot_resource" "my_resource" {
+  parent   = "tmod:@turbot/turbot#/"
+  type     = "tmod:@turbot/aws#/resource/types/account"
+  full_data     = <<EOT
+{
+  "foo": "bar",
```
Comment (kaidaguerre) [inline website/docs/r/resource.html.markdown]: have realistic properties in the example

**Review by kaidaguerre — CHANGES_REQUESTED** [review]

File: website/docs/r/resource.html.markdown [hunk 4]
```diff
@@ -35,16 +35,50 @@ EOT
...
 
+###Using full_data
+
+```hcl
+resource "turbot_resource" "my_resource" {
+  parent   = "tmod:@turbot/turbot#/"
+  type     = "tmod:@turbot/aws#/resource/types/account"
+  full_data     = <<EOT
+{
+  "type": "aws",
```
Comment (kaidaguerre) [inline website/docs/r/resource.html.markdown:46]: if it's full data  you should probably include account id

File: website/docs/r/resource.html.markdown [hunk 5]
```diff
@@ -35,16 +35,50 @@ EOT
...
+- `data` - (Optional) JSON representation of resource properties to be managed by Terraform. These must be valid for the type schema. NOTE: If additional properties are set on the resource by other means, they are ignored by Terraform.
+- `metadata` - (Optional) JSON representation of resource metadata properties to be managed by Terraform. NOTE: If additional metadata properties are set on the resource by other means, they are ignored by Terraform.
+- `full_data` - (Optional) JSON representation of all resource properties to be set on the resource. These must be valid for the type schema. NOTE: If additional properties are set on the resource by other means, they are removed.
+- `full_metadata` - (Optional) JSON representation of all resource metadata properties to be set on the resource. NOTE: If additional metadata properties are set on the resource by other means, they are removed.
 - `akas` - (Optional) Unique identifier of the resource.
 - `tags` - (Optional) User defined label for grouping resources.
+ 
+**NOTE**: You must not specify `data` and `full_data` OR `metadata` and `full_metadata` arguments together.
```
Comment (kaidaguerre) [inline website/docs/r/resource.html.markdown]: NOTE: only one of `data` and `full_data` may be specified. Likewise,  only one of `metadata ` and `full_metadata ` may be set.

**Review by kaidaguerre — CHANGES_REQUESTED** [review]

File: website/docs/r/resource.html.markdown (hunk 4 above)
Comment (RupeshPatil20) [reply website/docs/r/resource.html.markdown:46]: added Id in full_data

File: website/docs/r/resource.html.markdown [hunk 6]
```diff
@@ -35,16 +35,51 @@ EOT
...
+- `parent` - (Required) The identifier of the parent resource under which this resource will be created.
 - `type` - (Required) Defines the type of the resource to be created.
-- `data` - (Required) JSON representation of the details of the resource. When parsed, it must be valid for the `type` schema.
-- `metadata` - (Optional) A set of data that describes and gives information about the data of the resource.
+- `data` - (Optional) JSON representation of resource properties to be managed by Terraform. These must be valid for the type schema. NOTE: If additional properties are set on the resource by other means, they are ignored by Terraform.
+- `metadata` - (Optional) JSON representation of resource metadata properties to be managed by Terraform. NOTE: If additional metadata properties are set on the resource by other means, they are ignored by Terraform.
+- `full_data` - (Optional) JSON representation of all resource properties to be set on the resource. These must be valid for the type schema. NOTE: If additional properties are set on the resource by other means, they are removed.
```
Comment (kaidaguerre) [inline website/docs/r/resource.html.markdown]: "The data must be valid for the resource type schema"

File: website/docs/r/resource.html.markdown [hunk 7]
```diff
@@ -35,16 +35,51 @@ EOT
...
 ## Argument Reference
 
 The following arguments are supported:
 
-- `parent` - (Required) The `id` or `aka` of the level at which the Turbot resource will be created.
+- `parent` - (Required) The identifier of the parent resource under which this resource will be created.
 - `type` - (Required) Defines the type of the resource to be created.
-- `data` - (Required) JSON representation of the details of the resource. When parsed, it must be valid for the `type` schema.
-- `metadata` - (Optional) A set of data that describes and gives information about the data of the resource.
+- `data` - (Optional) JSON representation of resource properties to be managed by Terraform. These must be valid for the type schema. NOTE: If additional properties are set on the resource by other means, they are ignored by Terraform.
```
Comment (kaidaguerre) [inline website/docs/r/resource.html.markdown]: JSON only or yaml as well?

**Review by kaidaguerre — CHANGES_REQUESTED** [review]

File: website/docs/r/resource.html.markdown (hunk 7 above)
Comment (RupeshPatil20) [reply website/docs/r/resource.html.markdown]: Strictly JSON

**Pull request merged**